debian/import-sampleschool.1
.TH import-sampleschool 1
.SH NAME
import-sampleschool \- imports sample school data into SchoolTool
.SH SYNOPSIS
.B import-sampleschool
.I "[options]"
.SH DESCRIPTION
.B import-sampleschool
is a script that imports sample school data into a running SchoolTool server.
The server must be running, and it must not contain any data, or the import
will fail.
.SH OPTIONS
.TP
.B \-h HOSTNAME, \-\-host=HOSTNAME
Specify the hostname of the server (default: localhost).
.TP
.B \-p PORT, \-\-port=PORT
Specify the port number of the server (default: 7001).
.SH BUGS
The import script performs up to six HTTP requests per object imported and
is very slow.
.SH "SEE ALSO"
schooltool-server(1), schooltool-client(1), wxschooltool(1).


CHANGES.txt
=======
CHANGES
=======

2.6.4 (unreleased)
------------------

- Support CKEditor 4.x (https://bugs.launchpad.net/bugs/1309133)
- Delete schedules that reference a deleted timetable (https://bugs.launchpad.net/bugs/1280528)
- Improve tab style to make active tab more clear (https://launchpad.net/bugs/1023378)
- Timetables copied from active year retain old start/end dates (https://launchpad.net/bugs/1005962)
- Updated translations


2.6.3 (2014-04-12)
------------------

New features
++++++++++++

- Switch to CKEditor (https://bugs.launchpad.net/bugs/485898)
- Celery 3 support (https://bugs.launchpad.net/bugs/1276384)
- Latest translations from Launchpad

Bugfixes
++++++++

- Fixed exception dates format in calendar widget of recurring events (https://launchpad.net/bugs/372889)
- Fixed export of deleted timetables (https://launchpad.net/bugs/1281335)
- Updated copyright (https://launchpad.net/bugs/1289277)
- Fixed term/section navigation in a different school year (https://launchpad.net/bugs/1281050)


2.6.2 (2014-02-15)
------------------

- Fix upgrade from a fresh install of version 2.6.0 (https://launchpad.net/bugs/1258529)
- Fix delete year/term if it contains non-ascii chars.
- Updated translations


2.6.1 (2013-11-21)
------------------

- Synchronize timezone set in preferences with timetables and schedules (https://launchpad.net/bugs/1239468)
- Set initial timezone from ``/etc/timezone`` (https://launchpad.net/bugs/1030211)
- Add SchoolTool versions to the Server page.
- Person cannot see his own preferences (https://launchpad.net/bugs/1247825)
- Do not escape extra spaces in comments (https://launchpad.net/bugs/1241022)
- Fix error when trying to replace existing school logo (https://launchpad.net/bugs/1250733)
- Restore the warning about unsaved changes in gradebook, broken after adding
  autocompletion (https://launchpad.net/bugs/1253070)
- Updated translations


2.6.0 (2013-10-10)
------------------

- Don't reverse name order (https://launchpad.net/bugs/1201887)
- Add "Name sorting" option to Server settings, display the selected name
  column first in person tables.
- Person search does not find by full name (https://launchpad.net/bugs/894798)
- Change manager's full name to "Default Manager" (https://launchpad.net/bugs/1227220)
- Fix timetable wizard for weekly timetables with different periods order.
- Fix crash when displaying timetables with only one day (https://launchpad.net/bugs/1235150)
- Fix navigation inside the School tab if a non-active school year is selected,
  or school years title contains accented characters (https://launchpad.net/bugs/1234690)
- Setting multiple languages in schooltool.conf breaks remote reports (https://launchpad.net/bugs/1229013)
- Improvements to PDF grid and table style.
- Extra info in traceback page and submit to launchpad button
- Updated translations, removed unsupported ones (less than 25% translated).


2.5.3 (2013-09-17)
------------------

Bugfixes
++++++++

- Fix integer demographic fields handling in importer (https://launchpad.net/bugs/1211613)
- Add Activate school year link in School tab under Customize (https://launchpad.net/bugs/1216992 )
- Fix links and Done buttons inside School tab when a different school year is selected
- Add manager to Site Managers and Clerks groups (https://launchpad.net/bugs/1216915)
- Access setting: allow instructor to add students to their section (https://launchpad.net/bugs/637759)
- Access setting: allow instructor to modify their section (https://launchpad.net/bugs/683313)
- Access setting: instructors can schedule their own sections (https://launchpad.net/bugs/604643)
- Display contact information of a person in profile.
- Updated translations

Internals
+++++++++

- Add ability to disable views in customized packages (https://launchpad.net/bugs/716346)


2.5.2 (2013-08-22)
------------------

- Autocompletion in gradebook for discrete score systems.
- TAB key in gradebook moves to the next cell.
- Further updates to permissions.
- Clerks and administrators now have appropriate access to parts of the School.
- Update GPL license and file headers from gnu.org (https://launchpad.net/bugs/1211145)
- Fix failures when creating timetables with rotating periods and weekly time slots (https://launchpad.net/bugs/1214131)
- Updated translations


2.5.1 (2013-08-01)
------------------

- Differentiate permissions of administrators, clerks, and managers.
- Scripts and styles for the new attendance features.
- Updated translations


2.5.0 (2013-06-03)
------------------

Reports
+++++++

- Background XLS import (disabled)
- Background XLS export
- Background report generation
- Multiple reports in a .zip archive
- Messages about finished background tasks
- Added resource library for filtered/chart reports
- Removed Person XML reports

Server
++++++

- There are now 4 processes, controlled by ``supervisor``: ``schooltool``, ``zeo``,
  ``celery_report``, and ``redis``.
- New ``make`` targets: start, stop, restart, rerun, status.
- You have to install ``redis-server`` package on the system. Other new
  dependencies are Python packages.
- Instances have to be updated to run this version. If you run from a sandbox,
  do ``bin/make-schooltool-instance instance``.


2.4.1 (2013-05-30)
------------------

Reports
+++++++

- Fixed report generation to not do network requests (https://launchpad.net/bugs/1185418).
- Fixed ratio of school logo.
- Fixed rendering of "&" in headers.

Bugfixes
++++++++

- Warn about unsaved changes when leaving gradebook (https://launchpad.net/bugs/907999)
- Comment dialog submits the gradebook

Language
++++++++

- Replaced Filter by... labels
- Updated Italian, Greek, Portuguese, and other translations.


2.4.0 (2013-04-19)
------------------

- Fix crash on startup in 2.4b1 release
- Fix PDF reports crashing with newer z3c.rml (https://launchpad.net/bugs/1169723)
- Remove dependency on ``zope.app.server``
- Remove or deprecate unused schooltool.conf options

import-sampleschool.py
#!/usr/bin/env python
"""
A script to import sample school data into SchoolTool.
"""

import os
import sys
if sys.version_info < (2, 3):
    print >> sys.stderr, '%s: need Python 2.3 or later.' % sys.argv[0]
    print >> sys.stderr, 'Your python is %s' % sys.version
    sys.exit(1)

basedir = os.path.abspath(os.path.dirname(__file__))
datadir = basedir
sys.path.insert(0, os.path.join(basedir, 'src'))

# -- Do not remove this line --

import urllib
import getopt


class Error(Exception):
    pass


class SampleSchoolImporter:

    host = 'localhost'
    port = 7001

    expected_version = 'SchoolTool/0.4'

    datadir = datadir
    ttconfig_filename = datadir + '/ttconfig.data'

    def main(self, argv):
        """Generate and import sample school data."""
        try:
            self.process_args(argv)
            self.check_server_running()
            self.check_server_empty()
            self.import_csv_files()
            self.import_timetable_data()
        except Error, e:
            print >> sys.stderr, str(e)
            return 1
        else:
            return 0

    def process_args(self, argv):
        """Process command line arguments."""
        try:
            opts, args = getopt.getopt(argv[1:], 'h:p:', ['host=', 'port='])
        except getopt.error, e:
            raise Error(str(e))

        for k, v in opts:
            if k in ('-h', '--host'):
                self.host = v
            if k in ('-p', '--port'):
                try:
                    self.port = int(v)
                except ValueError, e:
                    raise Error("Invalid port number: %s" % v)

    def check_server_running(self):
        """Check that the server is running, and it is the correct version."""
        try:
            # urllib.urlopen uses FancyURLopener which is too fancy
            f = urllib.URLopener().open("http://%s:%s"
                                        % (self.host, self.port))
        except IOError:
            raise Error("SchoolTool server not listening on %s:%s"
                        % (self.host, self.port))
        else:
            version = f.info().getheader('Server')
            if version != self.expected_version:
                raise Error("Server version is %s, expected %s"
                            % (version, self.expected_version))

    def check_server_empty(self):
        """Check that the server is running, and it is the correct version."""
        for path in '/groups/teachers', '/ttschemas/default':
            try:
                # urllib.urlopen uses FancyURLopener which is too fancy
                f = urllib.URLopener().open("http://%s:%s%s"
                                            % (self.host, self.port, path))
            except IOError:
                # good, we got a 404
                return
            else:
                raise Error("SchoolTool server already has data imported."
                            " Remove your Data.fs if necessary,\n"
                            "restart the server and try again.")

    def import_csv_files(self):
        """Import data from CSV files."""
        from schooltool.clients.csvclient import CSVImporter, DataError
        os.chdir(self.datadir)
        importer = CSVImporter(host=self.host, port=self.port)
        try:
            importer.run()
        except DataError, e:
            raise Error(str(e))

    def import_timetable_data(self):
        """Import timetable data from ttconfig.data."""
        import schooltool.clients.client
        ttconfig = file(self.ttconfig_filename)
        c = schooltool.clients.client.Client(stdin=ttconfig)
        c.use_rawinput = False
        c.input_hook = lambda prompt: ttconfig.readline()[:-1]
        c.server = self.host
        c.port = self.port
        c.cmdloop()


if __name__ == '__main__':
    sys.exit(SampleSchoolImporter().main(sys.argv))
